File: AlphaBlending/AlphaBlending.py
import cv2
 
# Read the images
foreground = cv2.imread("puppets.png")
background = cv2.imread("ocean.png")
alpha = cv2.imread("puppets_alpha.png")
 
# Convert uint8 to float
foreground = foreground.astype(float)
background = background.astype(float)
 
# Normalize the alpha mask to keep intensity between 0 and 1
alpha = alpha.astype(float)/255
 
# Multiply the foreground with the alpha matte
foreground = cv2.multiply(alpha, foreground)
 
# Multiply the background with ( 1 - alpha )
background = cv2.multiply(1.0 - alpha, background)
 
# Add the masked foreground and background.
outImage = cv2.add(foreground, background)
 
# Display image
cv2.imshow("outImg", outImage/255)
cv2.waitKey(0)

File: AlphaBlending/AlphaBlending.cpp
#include opencv2/opencv.hpp
 
using namespace cv;
using namespace std;
 
int main(int argc, char** argv)
{
 
    Mat foreground = imread("puppets.png");
    Mat background = imread("ocean.png");
    Mat alpha = imread("puppets_alpha.png");
     
    foreground.convertTo(foreground, CV_32FC3);
    background.convertTo(background, CV_32FC3);
    alpha.convertTo(alpha, CV_32FC3, 1.0/255); 
 

    Mat ouImage = Mat::zeros(foreground.size(), foreground.type());
 

    multiply(alpha, foreground, foreground); 

    multiply(Scalar::all(1.0)-alpha, background, background); 
 

    add(foreground, background, ouImage); 

    imshow("alpha blended image", ouImage/255);
    waitKey(0);
     
    return 0;
}
